TemporaryParser-Tests/TPRBParserTest.class.st

```diff
@@ -168,104 +178,437 @@ TPRBParserTest >> testBlockReturnNode [
-	self compare: (TPRBMethodNode selector: #bar
-				body: (TPRBSequenceNode statements: (OrderedCollection 
-								with: (TPRBCascadeNode messages: (OrderedCollection with: messageNode with: messageNode)))))
-		to: (self parserClass parseMethod: 'bar self foo; foo')
+				arguments: (Array with: (TPRBLiteralNode value: 0))).
+]
+
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationSelfNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'self')
+		  to: (TPRBSelfNode new).
+]
+
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationSuperNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'super')
+		  to: (TPRBSuperNode new).
 ]
 
-{ #category : #'tests comparing' }
-TPRBParserTest >> testEqualToWithMapping [
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationThisContexNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'thisContext')
+		  to: (TPRBThisContextNode new).
+]
+
+{ #category : #testCreationProtocol }
+TPRBParserTest >> testCreationVariableNodeWithExpression [
+	self compare: (self parserClass parseExpression: 'variable')
+		  to: (TPRBVariableNode named: #variable start: 1).
+]
+
+{ #category : #testArray }
+TPRBParserTest >> testDelimitationsOfArrayNodesAreBrackets [
 	| tree |
-	tree := self treeWithEverything.
-	self assert: (tree equalTo: tree withMapping: Dictionary new).
-	tree := self treeWithReallyEverything.
-	self assert: (tree equalTo: tree withMapping: Dictionary new).
+	#('{}' '{.}' '{..}' '{foo. bar}' '{foo. bar.}' '{foo. bar. .}' '{. foo. bar}' '{foo.. bar}')
+	do: [ :each |  tree := self parserClass parseExpression: each.
+						self assert: tree left equals: 1.
+						self assert: tree right equals: each size ]
+]
+
+{ #category : #'error testing' }
+TPRBParserTest >> testFaultyLiteralRaiseSyntaxError [
+	#('#"bar"foo' '# foo' '#1' '#12' '#12.3' '# 1' '##1' '#"bar"1') 
+		do: [ :expr | 
+		self should: [self parserClass parseExpression: expr] raise: SyntaxErrorNotification ]
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageAssignementOfANonVariable [
+	"Raising a syntax exception when trying to do an assignement on a message."
+	self should: [self parserClass parseExpression: 'receiver selector := 1 + 2'] raise: SyntaxErrorNotification.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageAssignementWithMisplacedDot [
+	"The idea is for the parser to realise the first dot isn't supposed to be here and select it. Also add a side note to give suggestion     on the provenance of the error."
+	self should: [self parserClass parseExpression: 'temp := .1 + 2.'] raise: SyntaxErrorNotification.
+	self should: [self parserClass parseExpression: 'temp := 1. + 2.'] raise: SyntaxErrorNotification.
+	self should: [self parserClass parseExpression: 'temp := 1 +. 2.'] raise: SyntaxErrorNotification.

```

Comment:
And this one is yet a different one. It's a binary message without all its arguments.